python/pfs/drp/stella/readLineList.py

```diff
@@ -46,35 +48,43 @@ def run(self, detectorMap=None, metadata=None):
             Lines from the linelist.
         """
         lines = ReferenceLineSet.empty()
-        for filename in self.config.lineListFiles:
-            lines.extend(ReferenceLineSet.fromLineList(filename))
+        if self.config.lineListFiles:
+            for filename in self.config.lineListFiles:
+                lines.extend(ReferenceLineSet.fromLineList(filename))
         if self.config.restrictByLamps:
```

Comment:
I don't think `restrictByLamps` makes any sense in this new world. I think this line should simply be `else:`. That is to say, either the user specifies linelists or he relies on the header. I don't think there's a middle ground.